vhdl_syntax/src/parser/productions/names.rs

```diff
@@ -5,31 +5,506 @@
 // Copyright (c)  2025, Lukas Scheller lukasscheller@icloud.com
 
 use crate::parser::Parser;
-use crate::syntax::node_kind::NodeKind::{Label, Name};
+use crate::syntax::node_kind::NodeKind::*;
+use crate::tokens::Keyword as Kw;
 use crate::tokens::TokenKind::*;
 use crate::tokens::TokenStream;
 
-impl<T: TokenStream> Parser<T> {
-    pub fn designator(&mut self) {
-        self.expect_one_of_tokens([Identifier, StringLiteral, CharacterLiteral]);
+fn is_start_of_attribute_name<T: TokenStream>(parser: &mut Parser<T>) -> bool {
+    // Checking for `LeftSquare || Tick` will result in ambiguities with other grammar rules where a signature is possible right after a name.
+    // Those rules can be `alias_declaration` (LRM §6.6.1) and `subprogram_instantiation_declaration` (LRM §4.4).
+    // By checking whether the closing square bracket is followed by a `Tick` this ambiguity is resolved
+    match parser.peek_token() {
+        Some(Tick) => true,
+        Some(LeftSquare) => {
+            let mut idx = 1;
+            let mut bracket_count = 1;
+
+            while bracket_count > 0 {
+                match parser.peek_nth_token(idx) {
+                    Some(LeftSquare) => bracket_count += 1,
+                    Some(RightSquare) => bracket_count -= 1,
+                    Some(_) => {}
+                    None => {
+                        return false;
+                    }
+                }
+
+                idx += 1;
+            }
+
+            parser.next_nth_is(Tick, idx)
+        }
+        Some(_) | None => false,
     }
+}
 
+impl<T: TokenStream> Parser<T> {
     pub fn name(&mut self) {
+        // (Based on) LRM §8.1
+        // The LRM grammar rules for names were transformed to avoid left recursion.
+
+        // In contrast to the LRM, this parsing routine is greedy. Meaning, it will consume trailing parenthesized
+        // expressions even if the belong to an outer grammar rule!
         self.start_node(Name);
-        // TODO
-        self.designator();
+
+        if self.next_is(LtLt) {
+            self.external_name();
+        } else {
+            self.expect_one_of_tokens([Identifier, StringLiteral, CharacterLiteral]);
+        }
+
+        self.name_tail();
         self.end_node();
     }
 
     pub fn type_mark(&mut self) {
         self.name()
     }
 
-    pub fn opt_label(&mut self) {
+    pub(crate) fn designator(&mut self) {
+        // TODO: That designator is not fully LRM compliant
```

Comment:
Why not? What is missing?

Reply:
A `designator` as defined per LRM §4.2.1 should be either an `identifier` or an `operator_symbol` (character literal). It seems like `Parser::designator` got extended to also fit the `alias_designator` grammar rule.  

Other than that there are `actual_designator`, `attribute_designator`, `entity_designator` and `formal_designator` which partially overlap with the current implementation, but I do not think we should extend `Parser::designator` to be applicable for all these cases (especially since `actual_designator` can be really complex).